indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -547,6 +556,9 @@ public Void call() throws Exception
                   RemoteTaskRunnerWorkItem workItem = pendingTasks.remove(taskId);
                   taskComplete(workItem, null, TaskStatus.failure(taskId));
                 }
+                finally {
+                  tryAssignTasks.remove(taskId);
```

Comment:
I believe this shouldn't run if `tryAssignTasks.putIfAbsent` returned non-null, as it means some other thread is handling this task.